test/integration_tests/test_delete_executor.py

```diff
@@ -0,0 +1,166 @@
+# coding=utf-8
+# Copyright 2018-2022 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import unittest
+from test.util import create_sample_video, file_remove, load_inbuilt_udfs
+
+import numpy as np
+import pandas as pd
+
+from eva.catalog.catalog_manager import CatalogManager
+from eva.server.command_handler import execute_query_fetch_all
+from eva.catalog.catalog_type import ColumnType, IndexType, NdArrayType, TableType
+from eva.catalog.catalog_utils import xform_column_definitions_to_catalog_entries
+from eva.catalog.sql_config import IDENTIFIER_COLUMN
+from eva.configuration.configuration_manager import ConfigurationManager
+from eva.parser.create_statement import ColumnDefinition
+from eva.utils.generic_utils import generate_file_path
+from eva.models.storage.batch import Batch
+from eva.storage.storage_engine import StorageEngine
+
+
+class DeleteExecutorTest(unittest.TestCase):
+    def setUp(self):
+        # Bootstrap configuration manager.
+        ConfigurationManager()
+
+        # Reset catalog.
+        CatalogManager().reset()
+
+        load_inbuilt_udfs()
+
+        # Create feature vector table and raw input table.
+        id1 = 5
+        id2 = 15
+        id3 = 25
+
+        feat1 = np.array([[0, 0, 0]]).astype(np.float32)
+        feat2 = np.array([[100, 100, 100]]).astype(np.float32)
+        feat3 = np.array([[200, 200, 200]]).astype(np.float32)
+
+        input1 = np.array([[0, 0, 0]]).astype(np.uint8)
+        input2 = np.array([[100, 100, 100]]).astype(np.uint8)
+        input3 = np.array([[200, 200, 200]]).astype(np.uint8)
+
+        # Create table.
+        feat_col_list = [
+            ColumnDefinition("id", ColumnType.INTEGER, NdArrayType.UINT8, (1)),
+            ColumnDefinition("feat", ColumnType.NDARRAY, NdArrayType.FLOAT32, (1, 3)),
+            ColumnDefinition("input", ColumnType.NDARRAY, NdArrayType.UINT8, (1, 3)),
+        ]
+        feat_col_entries = xform_column_definitions_to_catalog_entries(feat_col_list)
+
+        input_col_list = [
+            ColumnDefinition("id", ColumnType.INTEGER, NdArrayType.UINT8, (1)),
+            ColumnDefinition("feat", ColumnType.NDARRAY, NdArrayType.FLOAT32, (1, 3)),
```

Comment:
You can consider multiple tables, some with primitive columns and others with ndarray-based columns.